controllers/machine_controller.go

```diff
@@ -183,9 +185,15 @@ func (r *MachineReconciler) reconcileDelete(ctx context.Context, cluster *cluste
 		}
 	} else {
 		klog.Infof("Deleting node %q for machine %q", m.Status.NodeRef.Name, m.Name)
-		if err := r.deleteNode(ctx, cluster, m.Status.NodeRef.Name); err != nil && !apierrors.IsNotFound(err) {
-			klog.Errorf("Error deleting node %q for machine %q: %v", m.Status.NodeRef.Name, m.Name, err)
-			return ctrl.Result{}, err
+		err = wait.PollImmediate(time.Second*2, time.Second*10, func() (bool, error) {
+			if err := r.deleteNode(ctx, cluster, m.Status.NodeRef.Name); err != nil && !apierrors.IsNotFound(err) {
+				r.Log.Error(err, "Error deleting node", "node", m.Status.NodeRef.Name, "machine", m.Name)
```

Comment:
Do we need to log this error given that we're retrying?

Reply:
No, we should only log once, at the end.